PayloadPkg/OsLoader/OsLoader.c

```diff
@@ -154,8 +143,129 @@ ParseIasImage (
     return EFI_UNSUPPORTED;
   }
 
+  return Status;
+}
+
+/**
+  Parse CONTAINER image
+
+  This function will parse CONTAINER image and get every file blobs info, Especially
+  get the command line file and ELF/kernel file for boot.
+
+  @param[in]      BootOption    Current boot option
+  @param[in, out] LoadedImage   Loaded Image information.
+
+  @retval  RETURN_SUCCESS       Parse IAS image successfully
+  @retval  Others               There is error when parsing IAS image.
+**/
+EFI_STATUS
+ParseContainerImage (
+  IN     OS_BOOT_OPTION      *BootOption,
+  IN OUT LOADED_IMAGE        *LoadedImage
+  )
+{
+  EFI_STATUS                  Status;
+  CONTAINER_HDR              *ContainerHdr;
+  UINT64                      ComponentName;
+  LOADER_COMPRESSED_HEADER   *LzHdr;
+  IMAGE_DATA                  File[MAX_IAS_SUB_IMAGE];
+  UINT8                       Index;
+
+  ContainerHdr = (CONTAINER_HDR  *)LoadedImage->IasImage.Addr;
+  if (ContainerHdr->Signature != BOOT_MAGIC_PATTERN) {
+    return EFI_UNSUPPORTED;
+  }
+  Status = RegisterContainer ((UINT32) ContainerHdr);
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_INFO, "Image given is not a valid CONTAINER image\n"));
+    return EFI_LOAD_ERROR;
+  }
+
+  AddMeasurePoint (0x4080);
+  ZeroMem (File, sizeof (File));
+
+  DEBUG ((DEBUG_INFO, "CONTAINER size = 0x%x, image type = 0x%x, # of components = %d\n", LoadedImage->IasImage.Size, ContainerHdr->ImageType, ContainerHdr->Count));
+
+  // Enumerate all components
+  Index = 0;
+  ComponentName = 0;
+  while (TRUE) {
+    Status = GetNextAvailableComponent (ContainerHdr->Signature, (UINT32 *)&ComponentName);
+    if (EFI_ERROR(Status) || ((UINT32)ComponentName == SIGNATURE_32 ('_', 'S', 'G', '_')) ) {
```

Comment:
Don't we already have SIGNATURE_32 ('_', 'S', 'G', '_') definition? A single definition would be better.